src/common/cockpitconf.c

```diff
@@ -21,92 +21,152 @@
 
 #include "cockpitconf.h"
 
-#include "cockpithash.h"
-
-#include "errno.h"
-
-static GHashTable *cockpit_conf = NULL;
-static GHashTable *cached_strvs = NULL;
-
-const gchar *cockpit_config_file = "cockpit.conf";
-const gchar *cockpit_config_dirs[] = { PACKAGE_SYSCONF_DIR, NULL };
-
-static gboolean
-load_key_file (const gchar *file_path)
+#include <assert.h>
+#include <ctype.h>
+#include <err.h>
+#include <errno.h>
+#include <libgen.h>
+#include <limits.h>
+#include <regex.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* #define DEBUG 1 */
+#if DEBUG
+#define debug(fmt, ...) (fprintf (stderr, "cockpitconf: " fmt "\n", ##__VA_ARGS__))
+#else
+#define debug(...)
+#endif
+
+
+typedef struct Entry {
+    char *section;
+    char *key;
+    char *value;
+    char **strv_cache;
+    char strv_delimiter;
+    struct Entry *next;
+} Entry;
+
+static bool cockpit_conf_loaded = false;
+static Entry *cockpit_conf = NULL;
+
+const char *cockpit_config_file = "cockpit.conf";
+const char *cockpit_config_dirs[] = { PACKAGE_SYSCONF_DIR, NULL };
+
+static char*
+strlower (char *s)
```

Comment:
the regexps accept non-ascii group/key names (which is not spec compliant) but this function assumes ascii.  you should fix one of those.

Reply:
Fixed both now.